### PR TITLE
httpsinkexporter: add support for metrics and filtering

### PR DESCRIPTION
**Pull request by owais**
- now supports metrics datatype.
- also supports filtering by name and attributes (metrics only)
- query URL has changed from `/` to `/spans` and `/metrics`
- fixed some other subtle bugs/issues

### CODE REVIEW

**Review by atoulme — APPROVED** [review]
Approved, I have reached out to the team for a second look however.

File: internal/exporter/httpsinkexporter/factory.go [hunk 1]
```diff
@@ -31,10 +31,12 @@ const (
 
 // NewFactory creates a factory for httpsink exporter.
 func NewFactory() exporter.Factory {
+	sinkFactory := newSinkFactory()
```
Comment (rmfitzpatrick) [inline internal/exporter/httpsinkexporter/factory.go:34]: Does this ensure there's only one sink regardless of desired endpoint? What if multiples are wanted?

```
httpsink/one:
  endpoint: localhost:12345
httpsink/two:
  endpoint: localhost:23456
```

Comment (owais) [reply internal/exporter/httpsinkexporter/factory.go:34]: This ensures there is one sink (http server) per exporter instance. Your example will start two sink servers. This was introduced so both metrics and spans start only one server per exporter instance. In the previous implementation they'd have tried to start the server twice for a single exporter instance.

Comment (owais) [reply internal/exporter/httpsinkexporter/factory.go:34]: You're right. There is a bug in this logic. Will fix shortly.

**Review by rmfitzpatrick — APPROVED** [review]

**Comment by owais** [comment]
Thanks @rmfitzpatrick. All passing now.

**Pull request merged**